### PR TITLE
Problem: Transaction signer could be empty (return signer address array)

### PR DESCRIPTION
**Pull request by vincentysc**
Issue: #685

Assumption: 
1. `pubkey` is in `state`
2. `pubkey` is not in the `signerInfo` when broadcasting the Tx
3. `pubkey` should be returned when searching it by signer address with the same node

Solution: parser return signer address array and `function ParseSignerInfosToTransactionSigners()` try to get pubkey by signer address if the `pubkey` is empty

More info: [possible signer reference table](https://github.com/crypto-com/chain-indexing/tree/bugfix/empty-multisend-signer/documentation/parser)

### CODE REVIEW

File: bootstrap/syncmanager.go [hunk 1]
```diff
@@ -26,6 +29,7 @@ const DEFAULT_MAX_RETRY_TIME = MAX_RETRY_TIME_ALWAYS_RETRY
 type SyncManager struct {
 	rdbConn              rdb.Conn
 	client               *tendermint.HTTPClient
```
Comment (davcrypto) [inline bootstrap/syncmanager.go:31]: please help to update the naming to distinguish different clients

File: usecase/parser/failedmsg_test.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ var _ = Describe("ParseMsgCommands", func() {
 
 			pm := usecase_parser_test.InitParserManager()
 
-			cmds, err := parser.ParseBlockTxsMsgToCommands(
+			cmds, possibleSignerAddress, err := parser.ParseBlockTxsMsgToCommands(
```
Comment (davcrypto) [inline usecase/parser/failedmsg_test.go:27]: `possibleSignerAddress` -> `possibleSignerAddresses`

File: usecase/parser/msg.go [hunk 3]
```diff
@@ -29,8 +29,9 @@ func ParseBlockTxsMsgToCommands(
 	blockResults *model.BlockResults,
 	accountAddressPrefix string,
 	stakingDenom string,
-) ([]command.Command, error) {
+) ([]command.Command, []string, error) {
 	commands := make([]command.Command, 0)
+	var address []string
```
Comment (davcrypto) [inline usecase/parser/msg.go:34]: `address` -> `addresses` for list

File: usecase/parser/block.go [hunk 4]
```diff
@@ -35,13 +37,7 @@ func ParseBlockToCommands(
...
 
 	if len(blockResults.TxsResults) > 0 {
-		transactionCommands, parseErr := ParseTransactionCommands(txDecoder, block, blockResults, accountAddressPrefix)
-		if parseErr != nil {
-			return nil, fmt.Errorf("error parsing transaction commands: %v", parseErr)
-		}
-		commands = append(commands, transactionCommands...)
-
-		msgCommands, parseErr := ParseBlockTxsMsgToCommands(
+		msgCommands, possibleSignerAddress, parseErr := ParseBlockTxsMsgToCommands(
```
Comment (davcrypto) [inline usecase/parser/block.go:40]: `possibleSignerAddress ` -> `possibleSignerAddresses`

File: usecase/parser/authinfo.go [hunk 5]
```diff
@@ -4,30 +4,49 @@ import (
...
 	"github.com/crypto-com/chain-indexing/internal/base64"
 	"github.com/crypto-com/chain-indexing/usecase/model"
 	"github.com/crypto-com/chain-indexing/usecase/parser/utils"
 )
 
 func ParseSignerInfosToTransactionSigners(
+	cosmosClient cosmosapp_interface.Client,
 	signerInfos []utils.SignerInfo,
 	accountAddressPrefix string,
+	possibleSignerAddress []string,
```
Comment (davcrypto) [inline usecase/parser/authinfo.go:18]: `possibleSignerAddress` -> `possibleSignerAddresses`

File: infrastructure/cosmosapp/test/account.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
+package infrastructure_cosmosapp_test
```
Comment (davcrypto) [inline infrastructure/cosmosapp/test/account.go:1]: Help to remove this file if there is no need anymore

**Comment by davcrypto** [comment]
Thanks @vincentysc ! Overall LGTM and just a few comments about the naming alignment. Ready to go once aligned.

File: bootstrap/syncmanager.go (hunk 1 above)
Comment (vincentysc) [reply bootstrap/syncmanager.go]: changed from `client` => `tendermintClient`.

**Review by davcrypto — APPROVED** [review]

**Pull request merged**